### DISCUSSION TITLE
installation problem

### DISCUSSION THREAD

**Discussion by Zil117** [frangoteam/FUXA#1428] [Q&A]
hello, is anyone can help me with the problem that im facing on the installation.

C:\Windows\System32>npm install -g --unsafe-perm @frangoteam/fuxa
npm warn deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated npmlog@5.0.1: This package is no longer supported.
npm warn deprecated npmlog@6.0.2: This package is no longer supported.
npm warn deprecated npmlog@4.1.2: This package is no longer supported.
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated are-we-there-yet@3.0.1: This package is no longer supported.
npm warn deprecated are-we-there-yet@2.0.0: This package is no longer supported.
npm warn deprecated are-we-there-yet@1.1.7: This package is no longer supported.
npm warn deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm warn deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm warn deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm warn deprecated gauge@4.0.4: This package is no longer supported.
npm warn deprecated gauge@3.0.2: This package is no longer supported.
npm warn deprecated gauge@2.7.4: This package is no longer supported.
npm warn deprecated node-opcua-client-crawler@2.123.0: This package is deprecated, please contact sterfive to acquire access to @sterfive/crawler revamped and improved module
npm warn cleanup Failed to remove some directories [
npm warn cleanup   [
npm warn cleanup     'C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam\\fuxa\\node_modules\\sqlite3',
npm warn cleanup     [Error: EBUSY: resource busy or locked, rmdir 'C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\sqlite3'] {
npm warn cleanup       errno: -4082,
npm warn cleanup       code: 'EBUSY',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam\\fuxa\\node_modules\\sqlite3'
npm warn cleanup     }
npm warn cleanup   ],
npm warn cleanup   [
npm warn cleanup     'C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam\\fuxa\\node_modules',
npm warn cleanup     [Error: EBUSY: resource busy or locked, rmdir 'C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\sqlite3'] {
npm warn cleanup       errno: -4082,
npm warn cleanup       code: 'EBUSY',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam\\fuxa\\node_modules\\sqlite3'
npm warn cleanup     }
npm warn cleanup   ],
npm warn cleanup   [
npm warn cleanup     'C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\node-gyp\test'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam\\fuxa\\node_modules\\node-gyp\\test'
npm warn cleanup     }
npm warn cleanup   ],
npm warn cleanup   [
npm warn cleanup     'C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam\\fuxa',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\js-sdsl'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam\\fuxa\\node_modules\\js-sdsl'
npm warn cleanup     }
npm warn cleanup   ]
npm warn cleanup ]
npm error code 1
npm error path C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\node-snap7
npm error command failed
npm error command C:\WINDOWS\system32\cmd.exe /d /s /c prebuild-install || node-gyp rebuild
npm error prebuild-install WARN install No prebuilt binaries found (target=20.18.0 runtime=node arch=x64 libc= platform=win32)
npm error gyp info it worked if it ends with ok
npm error gyp info using node-gyp@8.4.1
npm error gyp info using node@20.18.0 | win32 | x64
npm error gyp info find Python using Python version 3.13.0 found at "C:\Users\User\AppData\Local\Programs\Python\Python313\python.exe"
npm error gyp ERR! find VS
npm error gyp ERR! find VS msvs_version not set from command line or npm config
npm error gyp ERR! find VS VCINSTALLDIR not set, not running in VS Command Prompt
npm error gyp ERR! find VS could not use PowerShell to find Visual Studio 2017 or newer, try re-running with '--loglevel silly' for more details
npm error gyp ERR! find VS looking for Visual Studio 2015
npm error gyp ERR! find VS - not found
npm error gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
npm error gyp ERR! find VS
npm error gyp ERR! find VS **************************************************************
npm error gyp ERR! find VS You need to install the latest version of Visual Studio
npm error gyp ERR! find VS including the "Desktop development with C++" workload.
npm error gyp ERR! find VS For more information consult the documentation at:
npm error gyp ERR! find VS https://github.com/nodejs/node-gyp#on-windows
npm error gyp ERR! find VS **************************************************************
npm error gyp ERR! find VS
npm error gyp ERR! configure error
npm error gyp ERR! stack Error: Could not find any Visual Studio installation to use
npm error gyp ERR! stack     at VisualStudioFinder.fail (C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\node-gyp\lib\find-visualstudio.js:122:47)
npm error gyp ERR! stack     at C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\node-gyp\lib\find-visualstudio.js:75:16
npm error gyp ERR! stack     at VisualStudioFinder.findVisualStudio2013 (C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\node-gyp\lib\find-visualstudio.js:363:14)
npm error gyp ERR! stack     at C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\node-gyp\lib\find-visualstudio.js:71:14
npm error gyp ERR! stack     at C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\node-gyp\lib\find-visualstudio.js:384:16
npm error gyp ERR! stack     at C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\node-gyp\lib\util.js:54:7
npm error gyp ERR! stack     at C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\node-gyp\lib\util.js:33:16
npm error gyp ERR! stack     at ChildProcess.exithandler (node:child_process:430:5)
npm error gyp ERR! stack     at ChildProcess.emit (node:events:519:28)
npm error gyp ERR! stack     at maybeClose (node:internal/child_process:1105:16)
npm error gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:305:5)
npm error gyp ERR! System Windows_NT 10.0.22631
npm error gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam\\fuxa\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
npm error gyp ERR! cwd C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa\node_modules\node-snap7
npm error gyp ERR! node -v v20.18.0
npm error gyp ERR! node-gyp -v v8.4.1
npm error gyp ERR! not ok
npm error A complete log of this run can be found in: C:\Users\User\AppData\Local\npm-cache\_logs\2024-10-09T15_45_09_527Z-debug-0.log

**Comment by MatthewReed303** [comment]
@Zil117 try removing the snap7 package from package.json and try again, if you still have issues, use the docker version
https://github.com/frangoteam/FUXA/wiki/Installing-and-Running

**Comment by unocelli** [comment]
You are using node version 20!
Try to install the min version: `npm install -g --unsafe-perm @frangoteam/fuxa-min`

**Reply by Zil117** [reply]
C:\Users\User>npm install -g --unsafe-perm @frangoteam/fuxa-min
npm warn deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated npmlog@5.0.1: This package is no longer supported.
npm warn deprecated npmlog@6.0.2: This package is no longer supported.
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated are-we-there-yet@2.0.0: This package is no longer supported.
npm warn deprecated are-we-there-yet@3.0.1: This package is no longer supported.
npm warn deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm warn deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm warn deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm warn deprecated gauge@4.0.4: This package is no longer supported.
npm warn deprecated gauge@3.0.2: This package is no longer supported.
npm warn deprecated node-opcua-client-crawler@2.123.0: This package is deprecated, please contact sterfive to acquire access to @sterfive/crawler revamped and improved module
npm warn cleanup Failed to remove some directories [
npm warn cleanup   [
npm warn cleanup     'C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam\\fuxa-min\\node_modules\\@serialport',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa-min\node_modules\@serialport'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam\\fuxa-min\\node_modules\\@serialport'
npm warn cleanup     }
npm warn cleanup   ],
npm warn cleanup   [
npm warn cleanup     'C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam\\fuxa-min',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\Users\User\AppData\Roaming\npm\node_modules\@frangoteam\fuxa-min\node_modules\js-sdsl\dist\cjs\container'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\Users\\User\\AppData\\Roaming\\npm\\node_modules\\@frangoteam\\fuxa-min\\node_modules\\js-sdsl\\dist\\cjs\\container'
npm warn cleanup     }
npm warn cleanup   ]
npm warn cleanup ]
npm error code EEXIST
npm error path C:\Users\User\AppData\Roaming\npm\fuxa
npm error EEXIST: file already exists
npm error File exists: C:\Users\User\AppData\Roaming\npm\fuxa
npm error Remove the existing file and try again, or run npm
npm error with --force to overwrite files recklessly.
npm error A complete log of this run can be found in: C:\Users\User\AppData\Local\npm-cache\_logs\2024-10-23T03_14_07_214Z-debug-0.log
it seem it still have an error

**Reply by unocelli** [reply]
please check nodejs version? (supported 14, 16 or 18)
and check npm version too
